sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -765,7 +765,7 @@ temporalClause
 aggregationClause
     : GROUP BY groupingExpressionsWithGroupingAnalytics+=groupByClause
         (COMMA groupingExpressionsWithGroupingAnalytics+=groupByClause)*
-    | GROUP BY groupingExpressions+=expression (COMMA groupingExpressions+=expression)* (
+    | GROUP BY groupingExpressions+=namedExpression (COMMA groupingExpressions+=namedExpression)* (
```

Comment:
hmm isn't it a breaking change for normal SQL? The group by expression must be named now?

Reply:
Good question, each GROUP BY expression doesn't have to be named, it just has the option now :)

```
namedExpression
    : expression (AS? (name=errorCapturingIdentifier | identifierList))?
    ;
```

The `AS <alias>` or `<alias>` part is optional. So if it is absent, the behavior is the same as before. Otherwise if it is present, then we get an `Alias` expression whose child is the grouping key.

This should have no change in the query execution behavior for the aggregation operator, since the evaluation of the `Alias` expression is equivalent to the evaluation of its child expression:

```
  override def eval(input: InternalRow): Any = child.eval(input)
```

Reply:
@dtenedor this could be a breaking change for normal group by clause.
Given a table t1 with two columns `i` & `j`, the following invalid query will be allowed in the parser after changes:
```
select max(i) from t1 group by j as i
```

Reply:
No problem, it seems we prefer to prevent the grouping key aliases in non-pipe GROUP BY clauses. I pushed a commit to check for this, and only allow these aliases in SQL pipe operator GROUP BY clauses instead.

Reply:
@dtenedor could you explain why we need to support alias after the `GROUP BY` clause? It seems that the following SQL will still be confusing:
```
| AGGREGATE max(i) GROUP BY j as i
```

Reply:
Sure, good question. So the answer is that with SQL pipe semantics, the AGGREGATE operator's output attribute comprise all the grouping expressions first (in order), then all the aggregate functions next (in order). This is the specification from the original research paper [1].

The idea is that the list of expressions after the AGGREGATE keyword do not need to include any of the grouping keys; they may only comprise aggregate function calls. Then following operators can refer to the grouping keys (possibly using aliases) and/or aggregate functions as needed, including just `|> SELECT`ing a subset.

So this example:

```
| AGGREGATE max(i) GROUP BY j + 1 as i
```

would be equivalent to this using classic SQL:

```
SELECT j + 1 as i, max(i) FROM t
GROUP BY j + 1
```

<img width="905" alt="image" src="https://github.com/user-attachments/assets/de8c0bf2-7c33-4ac7-a946-d4f93162fc71">

[1] https://research.google/pubs/sql-has-problems-we-can-fix-them-pipe-syntax-in-sql/

Reply:
In classic SQL, the SELECT list can perform operations on the group by columns, e.g. `SELECT a + b FROM t GROUP BY a, b`, is it allowed in pipe? `|> AGGREGATE a + b GROUP BY a, b`, or people have to do `|> AGGREGATE GROUP BY a, b |> SELECT a + b`?

Reply:
Good question, this is true. In the pipe syntax, in order to perform projections on the grouping expressions, it is necessary to add a follow-up `|> SELECT` step like your example.